frontend/apps/pablo/defi/hooks/swaps/useSwaps.ts

```diff
@@ -224,61 +229,68 @@ export function useSwaps({
           ? baseAsset
           : quoteAsset;
       if (sideUpdated === "quote") {
-        const toAmount = calculateOutGivenIn(
+        const feeCharged = (
+          amount.gt(balance1) ? balance1 : amount
+        ).multipliedBy(feePercentage / 100);
+        const amountIn = (amount.gt(balance1) ? balance1 : amount).minus(
```

Comment:
I recommend moving this stuff to wherever the original equation is. Fee is part of the equations. Also please make sure to comment on things where the equation was found.

Reply:
Latest ref on the on-chain logic is at https://github.com/ComposableFi/composable/blob/main/rfcs/0008-pablo-lbp-cpp-restructure.md#41-fee-math-updates